### PR TITLE
add pragma printf and scanf annotations to stdio.d

### PR DESCRIPTION
**Pull request by WalterBright**
Now that the annotations are ignored per https://github.com/dlang/dmd/pull/11030 we can add them.

### CODE REVIEW

**Comment by WalterBright** [comment]
@jacob-carlborg looks like dlp-test-application needs an update?

```
Running ./dlp-test-application
--
  | /var/lib/buildkite-agent/builds/ci-agent-68ef3a9c-5d12-453f-8e0a-78cd3f822dd9-1/dlang/druntime/build/distribution/imports/core/stdc/stdio.d(1243,5): Error: unrecognized `pragma(printf)`
```

File: src/core/stdc/stdio.d [hunk 1]
```diff
@@ -1367,13 +1395,17 @@ else version (CRuntime_Microsoft)
...
     ///
+    pragma(printf)
     int  snprintf(scope char* s, size_t n, scope const char* format, ...);
 
     ///
+    pragma(printf)
     int _vsnprintf(scope char* s, size_t n, scope const char* format, va_list arg);
     ///
+    pragma(printf)
     int  vsnprintf(scope char* s, size_t n, scope const char* format, va_list arg);
```
Comment (Geod24) [inline src/core/stdc/stdio.d:1409]: @WalterBright : In this case, how will the compiler know to check the 3rd argument ?

**Comment by WalterBright** [comment]
@CyberShadow can you please look into the DAutoTest failure, please?

File: src/core/stdc/stdio.d (hunk 1 above)
Comment (WalterBright) [reply src/core/stdc/stdio.d:1409]: The last 2 parameters are always the same for the "v" functions.

Comment (Geod24) [reply src/core/stdc/stdio.d:1409]: True, but you still rely on a convention. Is it the last `char*` argument, or the argument named format, or some other rule ?

**Comment by jacob-carlborg** [comment]
> jacob-carlborg looks like dlp-test-application needs an update?

I'll update it.

File: src/core/stdc/stdio.d (hunk 1 above)
Comment (WalterBright) [reply src/core/stdc/stdio.d:1409]: The last one is `va_arg`, the second to last is `const(char)*`.

**Comment by WalterBright** [comment]
Blocking https://github.com/dlang/dmd/pull/11034

**Comment by jacob-carlborg** [comment]
> jacob-carlborg looks like dlp-test-application needs an update?

I've pushed an update and triggered a new build.

**Comment by WalterBright** [comment]
@jacob-carlborg thank you!

**Comment by WalterBright** [comment]
All green. Someone please approve!

**Comment by WalterBright** [comment]
> I'm pretty certain you're going to end up needing to give the argument index / name as GCC does...

We'll see. I did leave the design open to adding that if it becomes necessary. In the meantime, I'm biased towards doing things as simple as possible.

**Pull request merged**